### PR TITLE
Implement `bot.get_application_command`

### PR DESCRIPTION
**Pull request by Dorukyum**
## Checklist
- [x] If code changes were made then they have been tested.
    - [x] I have updated the documentation to reflect the changes.
- [ ] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, typehinting, examples, ...)

### CODE REVIEW

**Review by Lulalaby — DISMISSED** [review]

**Comment by CodeWithSwastik** [comment]
Would be better to use command name here otherwise its redundant as you can just `bot.application_commands[id]`.

**Review by FrostByte266 — DISMISSED** [review]
The behavior mentioned in the docstring and the actual behavior differ. The docs say that it will return `None` if no command is found, however, the `dict.get` method will raise an exception if the item is not found, unless a default value is provided.

**Comment by CodeWithSwastik** [comment]
> The behavior mentioned in the docstring and the actual behavior differ. The docs say that it will return `None` if no command is found, however, the `dict.get` method will raise an exception if the item is not found, unless a default value is provided.

That's wrong, it will return None

**Comment by Dorukyum** [comment]
> Would be better to use command name here otherwise its redundant as you can just `bot.application_commands[id]`. 

As we already discussed, multiple commands can have the same name. I can't say anything about this being redundant though.

**Comment by FrostByte266** [comment]
> > The behavior mentioned in the docstring and the actual behavior differ. The docs say that it will return `None` if no command is found, however, the `dict.get` method will raise an exception if the item is not found, unless a default value is provided.
> 
> That's wrong, it will return None

That's right, my bad. I forgot that it has a default value of None even if you don't explicitly specify that.

**Comment by FrostByte266** [comment]
> > Would be better to use command name here otherwise its redundant as you can just `bot.application_commands[id]`.
> 
> As we already discussed, multiple commands can have the same name. I can't say anything about this being redundant though.

I think it would still be beneficial, some users are used to calling a method as opposed to accessing a dictionary. Giving users a choice on which style they like won't hurt anything in my opinion.

**Comment by CodeWithSwastik** [comment]
> > Would be better to use command name here otherwise its redundant as you can just `bot.application_commands[id]`.
> 
> As we already discussed, multiple commands can have the same name. I can't say anything about this being redundant though.

This function would be useless if it is using ids to lookup as I've said before. 
As for command names being same, a second parameter can be added for which guild_id to get the command from and if its not given it will return the global command.

**Comment by CodeWithSwastik** [comment]
> I think it would still be beneficial, some users are used to calling a method as opposed to accessing a dictionary. Giving users a choice on which style they like won't hurt anything in my opinion.

It's just redundant, and using the command name is

**Comment by Middledot** [comment]
> > > Would be better to use command name here otherwise its redundant as you can just `bot.application_commands[id]`.
> > 
> > 
> > As we already discussed, multiple commands can have the same name. I can't say anything about this being redundant though.
> 
> This function would be useless if it is using ids to lookup as I've said before. As for command names being same, a second parameter can be added for which guild_id to get the command from and if its not given it will return the global command.

But there are also different types of application commands which can have the same name

**Comment by Middledot** [comment]
Maybe a method that takes string and AppCmdType enum as well as a guild_id

**Comment by CodeWithSwastik** [comment]
> Maybe a method that takes string and AppCmdType enum as well as a guild_id

That'd be alright and it can default to SlashCommands

**Comment by Dorukyum** [comment]
I will make it take `name`, `type` and `guild_ids` then.

**Comment by BobDotCom** [comment]
> This function would be useless if it is using ids to lookup as I've said before. As for command names being same, a second parameter can be added for which guild_id to get the command from and if its not given it will return the global command.
In my opinion it would be best to make `bot._application_commands` a dict, and `bot.application_commands` a list with a property getter and setter. This would be consistent with bot.commands and would allow for the faster lookup as internal methods would use the private variable. Regardless, the faster comprehension isn't substantial as I did some testing and found that both dicts and lists have a sub 1 millisecond lookup time with 110 commands (the max). Even so, it's possible someone could have thousands of guild commands, in which case (with 10 thousand) list lookup can take 5-15 seconds while dict lookup is steadily 1 millisecond.

**Comment by Lulalaby** [comment]
Pretty good idea

File: discord/bot.py [hunk 1]
```diff
@@ -136,7 +140,44 @@ def remove_application_command(
             The command that was removed. If the name is not valid then
             ``None`` is returned instead.
         """
-        return self.application_commands.pop(command.id)
+        return self._application_commands.pop(command.id)
+
+    def get_command(
```
Comment (BobDotCom) [inline discord/bot.py:145]:
```suggestion
    @property
    def get_command(self):
        """Shortcut for :meth:`.get_application_command`.
        
        .. note::
            Overridden in :class:`ext.commands.Bot`.
        
        .. versionadded:: 2.0
        """
        # TODO: Do something like we did in self.commands for this
        return self.get_application_command
    def get_application_command(
```

**Review by BobDotCom — CHANGES_REQUESTED** [review]

**Review by BobDotCom — APPROVED** [review]

**Review by Lulalaby — APPROVED** [review]

**Pull request merged**